packages/devp2p/src/util.ts

```diff
@@ -11,17 +10,17 @@ export const devp2pDebug = createDebugLogger('devp2p')
 
 export function keccak256(...buffers: Buffer[]) {
   const buffer = Buffer.concat(buffers)
-  return createKeccakHash('keccak256').update(buffer).digest()
+  return Buffer.from(_keccak256(buffer))
 }
 
 export function genPrivateKey(): Buffer {
-  const privateKey = randomBytes(32)
-  return privateKeyVerify(privateKey) ? privateKey : genPrivateKey()
+  const privateKey = utils.randomPrivateKey()
+  return utils.isValidPrivateKey(privateKey) ? Buffer.from(privateKey) : genPrivateKey()
 }
 
 export function pk2id(pk: Buffer): Buffer {
   if (pk.length === 33) {
-    pk = Buffer.from(publicKeyConvert(pk, false))
+    pk = Buffer.from(getPublicKey(pk, false))
```

Comment:
This method took 33 or 65-byte public key and converted it to 65-byte decompressed form. Now, it would treat `pk` as `private key` and convert it to `public key`. And private keys cannot be 33-byte, so there's an error in any case.

Reply:
There's `publicKeyConvert` method inside secp256k1-compat